src/api.ts

```diff
@@ -738,15 +738,16 @@ export class APIGatewayVTL extends VTL {
       } else if (isReferenceExpr(expr.expr) || isThisExpr(expr.expr)) {
         const ref = expr.expr.ref();
         if (ref === Number) {
-          return this.exprToJson(expr.args[0]);
+          // Number() = 0
+          return expr.args[0] ? this.exprToJson(expr.args[0]) : "0";
         } else {
           throw new SynthError(
             ErrorCodes.Unexpected_Error,
             "Called references are expected to be an integration."
           );
         }
       } else if (isIdentifier(expr.expr) && expr.expr.name === "Number") {
-        return this.exprToJson(expr.args[0]);
+        return expr.args[0] ? this.exprToJson(expr.args[0]) : "0";
```

Comment:
Looks like we have two ways of dealing with Number()? Do we need both?

Reply:
I don't think so? Actually not sure which one, you change may have changed it back to not be a ref.

Reply:
Apparently `Number` is still a ref. removed the second one.